### ISSUE TITLE
error while taking checkpoints

### ISSUE THREAD

**Issue by zimingH** [agsiddharth/CAL-DPDK-GEM5#1]
Hi,
I encountered some issue while running the code. I have completed the buildroot part for generating the kernel and disk images, and I have built the gem5.fast and gem5.opt. I used the docker to build the kernel and disks images, and then build the gem5 outside the docker. 

Following the guide in readme.md, I tried to run this line:
"./l2fwd-ckp.sh --take-checkpoint --num-nics 1 --script dpdk-1.sh" (I run this project's root directory, so the actual command for me is "run-scripts/l2fwd-ckp.sh --take-checkpoint --num-nics 1 --script dpdk-1.sh"). 
Then I get this bus error:

...
build/ARM/arch/arm/insts/pseudo.cc:173: warn:   instruction 'csdb' unimplemented
build/ARM/arch/arm/insts/pseudo.cc:173: warn:   instruction 'bti' unimplemented
Writing checkpoint
build/ARM/sim/simulate.cc:107: info: Entering event queue @ 618409311000.  Starting simulation...
run-scripts/l2fwd-ckp-fastsim.sh: line 23: 1751325 Bus error               (core dumped) "$GEM5_DIR/build/ARM/gem5.$GEM5TYPE" $DEBUG_FLAGS --outdir="$RUNDIR" "$GEM5_DIR"/configs/example/fs.py --cpu-type=$CPUTYPE --kernel="$RESOURCES/vmlinux" --disk="$RESOURCES/rootfs.ext2" --bootloader="$RESOURCES/boot.arm64" --root=/dev/sda --num-cpus=2 --mem-size=8192MB --script="$GUEST_SCRIPT_DIR/$GUEST_SCRIPT" --num-nics="$num_nics" --num-loadgens="$num_nics" --checkpoint-dir="$CKPT_DIR" $CONFIGARGS

I went further by switching the gem5.fast into gem5.opt when taking the checkpoints, and the following showed up:
0x00005555572cd5cf in gem5::Flags<unsigned short>::isSet (mask=2, this=0x2010001000022) at build/ARM/base/flags.hh:83
83          bool isSet(Type mask) const { return (_flags & mask); }
 
I have checked the $GIT_ROOT and it is correctly set to my repo root dir. Not sure what caused this.

Thanks,
Ziming

**Comment by agsiddharth** [comment]
Hi,

This could be a filesystem issue on the host system or some potential error in building the disk image. 

Have you tried running using the prebuilt disk images in the repo? (Clone using git LFS)
Additionally, can you let me know what system you are running this on?

**Comment by zimingH** [comment]
I tried using the prebuild disk image using the git LFS just now. Similar error is reported.

"
Writing checkpoint
build/ARM/sim/simulate.cc:107: info: Entering event queue @ 663382179000.  Starting simulation...
run-scripts/l2fwd-ckp.sh: line 22: 1037098 Bus error               (core dumped) "$GEM5_DIR/build/ARM/gem5.$GEM5TYPE" $DEBUG_FLAGS --outdir="$RUNDIR" "$GEM5_DIR"/configs/example/fs.py --cpu-type=$CPUTYPE --kernel="$RESOURCES/vmlinux" --disk="$RESOURCES/rootfs.ext2" --bootloader="$RESOURCES/boot.arm64" --root=/dev/sda --num-cpus=2 --mem-size=8192MB --script="$GUEST_SCRIPT_DIR/$GUEST_SCRIPT" --num-nics="$num_nics" --num-loadgens="$num_nics" --checkpoint-dir="$CKPT_DIR" $CONFIGARGS
"

I have attached the generated system log (the one in rundir/1NIC-ckp by default). 
[system.txt](https://github.com/agsiddharth/CAL-DPDK-GEM5/files/14913265/system.txt)

The Linux version is: VERSION="20.04.4 LTS (Focal Fossa)"

**Comment by Tomas-mr** [comment]
i got the same error when i tried to use the prebuild disk image using the git LFS as zimingH just reported.

`
build/ARM/sim/simulate.cc:107: info: Entering event queue @ 663382179000.  Starting simulation...
./l2fwd-ckp.sh: line 22: 248644 Bus error               (core dumped) "$GEM5_DIR/build/ARM/gem5.$GEM5TYPE" $DEBUG_FLAGS --outdir="$RUNDIR" "$GEM5_DIR"/configs/example/fs.py --cpu-type=$CPUTYPE --kernel="$RESOURCES/vmlinux" --disk="$RESOURCES/rootfs.ext2" --bootloader="$RESOURCES/boot.arm64" --root=/dev/sda --num-cpus=2 --mem-size=8192MB --script="$GUEST_SCRIPT_DIR/$GUEST_SCRIPT" --num-nics="$num_nics" --num-loadgens="$num_nics" --checkpoint-dir="$CKPT_DIR" $CONFIGARGS
`

**Comment by agsiddharth** [comment]
are you running the simulation under docker?

**Comment by Tomas-mr** [comment]
> are you running the simulation under docker?

yes, i running the simulation under docker, and the Linux version is ubuntu 20.04
